src/player.cpp

```diff
@@ -12228,26 +12228,36 @@ int player::encumb( body_part bp ) const
 int player::encumb(body_part bp, double &layers, int &armorenc) const
 {
     int ret = 0;
-    int layer[MAX_CLOTHING_LAYER] = { };
+    // First is the total of encumbrance on the given layer,
+    // and second is the highest encumbrance of any one item on the layer.
+    std::array<std::pair<int, int>, MAX_CLOTHING_LAYER> layer = { };
```

Comment:
In my compiler, changing it to
```
std::array<std::pair<int, int>, MAX_CLOTHING_LAYER> layer = {{
    {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}
}};
```

Did the job, but I'm not sure if it wouldn't cause problems in other compilers.